src/app/api/user/route.ts

```diff
@@ -0,0 +1,61 @@
+import { response200, response400, response401 } from "@/app/utils";
+import { z } from "zod";
+import { selectUserById, updateUser } from "@/model/UserModel";
+import getUserServerSide from "@/services/server/UserService";
+import {
+  ProfileGender,
+  ProfileGoals,
+  ProfileMakeupBrands,
+  ProfileProductsUsed,
+  ProfileRoutineProducts,
+  ProfileSkinConcern,
+  ProfileSkinConditions,
+  ProfileSkinType,
+  ProfileSkincareBrands,
+} from "@/services/types";
+
+const requestType = z.object({
+  gender: z.nativeEnum(ProfileGender),
+  birthdate: z.coerce.date().optional(),
+  goals: z.nativeEnum(ProfileGoals).array().optional(),
+  skin_types: z.nativeEnum(ProfileSkinType).array().optional(),
+  skin_concerns: z.nativeEnum(ProfileSkinConcern).array().optional(),
+  skin_conditions: z.nativeEnum(ProfileSkinConditions).array().optional(),
+  routine_products: z.nativeEnum(ProfileRoutineProducts).array().optional(),
+  makeup: z.boolean().optional(),
+  products_used: z.nativeEnum(ProfileProductsUsed).array().optional(),
+  skincare_brands: z.nativeEnum(ProfileSkincareBrands).array().optional(),
+  makeup_brands: z.nativeEnum(ProfileMakeupBrands).array().optional(),
+});
```

Comment:
We have the kinda the same TS definitions in 2 places. Here and also in the `updateUser` function.

If we're not careful they may get out of sync. So, best thing is to use the same definition for both. This way, they won't get out of sync.

So, let's:

1. Extract this zod object into a separate file. For example in `types.ts` and name if `ZodUserProfile`
2. Create a new type from it like this:

```
type UserProfile = z.infer<typeof ZodUserProfile>;
```

3. Here we'll import the Zod object and use it in the validation
4. Inside the `updateUser` function we'll use the `UserProfile` type for our function definition.